### PR TITLE
KBV-181 add browser test framework

### PR DESCRIPTION
**Pull request by MattBidewell**
### What changed

Adding testing framework for browser based tests

### Why did it change

As developers, we would like increased confidence the changes we perform dont create regressions in for the users.
By providing the cucumber testing framework to the project, we can increase confidence and reduce risk of errors with our changes.

### Issue tracking

- [KBV-181](https://govukverify.atlassian.net/browse/KBV-181)

### Other considerations

- [x] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,12 +12,42 @@ Clone this repository and then run
...
+
+This can be run by using:
+
+`yarn run mock`
+
+The frontend can be configure to use this server through changing two environment variables:
+
+- `NODE_ENV = development` - this enables a midldeware that passes the `x-scenario-id` header from web requests through to the API.
+- `API_BASE_URL = http://localhost:8090` - this points the frontend to the wiremock instance.
+- `ORDNANCE_API_URL = http://localhost:8090` - this points to an external API for address lookup. <!-- TODO: Move Ordnance to BE -->
```
Comment (MattBidewell) [inline README.md:38]: We can use the same wiremock instance temporarily

File: test/application/support/env.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+const { setWorldConstructor } = require("@cucumber/cucumber");
+
+require("playwright");
+
+const users = {
+  "Authenticatable Anita": {
```
Comment (merlinc) [inline test/application/support/env.js:6]: I don't think you need these authentication users, but you might want a Not Found Nigel or a Postcode Mismatch Mary in the future when you implement various error paths.

Again, for now, it's syntactic sugar to make it easier to remember what the result should be, as opposed to a generic "I" or "They"

File: src/app.js [hunk 3]
```diff
@@ -1,6 +1,8 @@
 const { PORT, SESSION_SECRET } = require("./lib/config");
 const { setup } = require("hmpo-app");
 
+const scenarioHeaders = require("./lib/scenario-headers");
```
[the remark this answers is not in the document]
Comment (MattBidewell) [reply src/app.js:4]: @merlinc 

I've made a seperate PR to change this as the PR was getting chunky!

https://github.com/alphagov/di-ipv-cri-address-front/pull/32

File: test/application/features/address.feature [hunk 4]
```diff
@@ -0,0 +1,14 @@
+@mock-api:address-success @success
+Feature: Happy path
+  Viewing the Address lookup successfully
+
+  Background:
+    Given Authenticatable Anita is using the system
+
+  Scenario: Selecting an address
+    Given they have started the address journey
+    Given they searched for their postcode "T35T1N"
```
[the remark this answers is not in the document]
Comment (MattBidewell) [reply test/application/features/address.feature:10]: I believe I've now resolved this 👀

File: test/application/pages/search.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  constructor(page) {
+    this.page = page;
+    this.url = "http://localhost:5010/search";
+  }
+
+  async getPageTitle() {
+    return await this.page.textContent("#header");
+  }
+
+  async searchPostcode(postcode = "TE5T1NG") {
```
Comment (MattBidewell) [inline test/application/pages/search.js:14]: Default to a test postcode, if no input. Is this needed?

**Review by merlinc — DISMISSED** [review]
LGTM

There's a few questions about how defaults for searches, and about the cucumber tests - let's see what we can improve once we've reviewed the address flow, and we've compared this against the KBV repo - no reason to hold up this PR for now

**Review by merlinc — APPROVED** [review]

**Pull request merged**